packages/dds/tree/src/tableSchema.ts

```diff
@@ -96,7 +118,42 @@ export namespace TableSchema {
 			IColumn<TPropsSchema> &
 			WithType<ScopedSchemaName<Scope, "Column">>;
 
-		type ColumnInsertableType = InsertableObjectFromSchemaRecord<typeof columnFields>;
+		// Note: ideally this type would just leverage `InsertableObjectFromSchemaRecord<typeof columnFields>`,
+		// but that results in broken `.d.ts` output due to a TypeScript bug.
+		// Instead we extract and inline the typing of the "props" field here, which seems to sufficiently work around the issue.
+		// type ColumnInsertableType = InsertableObjectFromSchemaRecord<typeof columnFields>;
+		type ColumnInsertableType = InsertableObjectFromSchemaRecord<
+			typeof columnFieldsBuiltInParts
+		> &
+			(FieldHasDefault<TPropsSchema> extends true
+				? {
+						/**
+						 * The column's properties.
```

Comment:
Docs need to live here in order for them to appear in intellisense for insertion code. Since `@inheritDoc` doesn't work with intellisense, I opted to duplicate with the base interface ☹️.

Reply:
I think this information should be in a privateRemartks, not a code review comment sine people editing this code in the future will need to know it.

Reply:
Fair point